### PR TITLE
Minor patches and fixes

### PR DESCRIPTION
**Pull request by clementfaure**
Hello!
Here are small patches for i.MX platforms, CAAM driver and a small fix in PKCS11 TA.

Is it worth it to add a line in the CI for 8590cb228 ?

### CODE REVIEW

**Comment by clementfaure** [comment]
@jforissier I'm not sure about the checkpatch error ... is it a false warning?

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

> Is it worth it to add a line in the CI for [8590cb2](https://github.com/OP-TEE/optee_os/commit/8590cb2284915869f16cd96c9f583c8dc35e37cf) ?

It certainly wouldn't hurt. Maybe add it for QEMUv8, it would have caught the issue as well I believe, and QEMUv8 is lacking a "full trace" build contrary to QEMU (v7) a few lines above:
```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index d348565db..a94cb7b91 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -119,6 +119,7 @@ jobs:
           _make CFG_WITH_USER_TA=n
           _make PLATFORM=vexpress-qemu_armv8a
           _make PLATFORM=vexpress-qemu_armv8a COMPILER=clang
+          _make PLATFORM=vexpress-qemu_armv8a CFG_TEE_CORE_LOG_LEVEL=4 CFG_TEE_CORE_DEBUG=y CFG_TEE_TA_LOG_LEVEL=4 CFG_CC_OPT_LEVEL=0 CFG_DEBUG_INFO=y
           _make PLATFORM=vexpress-qemu_armv8a CFG_TEE_CORE_LOG_LEVEL=0 CFG_TEE_CORE_DEBUG=n CFG_TEE_TA_LOG_LEVEL=0 CFG_DEBUG_INFO=n
           _make PLATFORM=vexpress-qemu_armv8a CFG_WITH_PAGER=y
           _make PLATFORM=vexpress-qemu_armv8a CFG_FTRACE_SUPPORT=y CFG_ULIBS_MCOUNT=y CFG_ULIBS_SHARED=y
```

**Comment by jforissier** [comment]
> @jforissier I'm not sure about the checkpatch error ... is it a false warning?

Yes, when code lines are a bit long they easily overflow the commit description max width. Let's ignore it, I prefer to have exact diagnotics messages it's good for searching.

**Comment by jforissier** [comment]
"ci: qemuv8: build with maximum log level" is:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with include ordering fixed in _caam_jr.c_.

For commit "ci: qemuv8: build with maximum log level", if willing to build with most debug features enabled, maybe consider (some of) the followings: `CFG_TEE_TA_MALLOC_DEBUG=y CFG_TEE_TA_MALLOC_DEBUG=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_TA_STACK_PROTECTOR_ALL=y` `CFG_TA_STATS=y`.

**Comment by clementfaure** [comment]
> `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with include ordering fixed in _caam_jr.c_.
> 
> For commit "ci: qemuv8: build with maximum log level", if willing to build with most debug features enabled, maybe consider (some of) the followings: `CFG_TEE_TA_MALLOC_DEBUG=y CFG_TEE_TA_MALLOC_DEBUG=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_TA_STACK_PROTECTOR_ALL=y` `CFG_TA_STATS=y`.

Well, I don't have a strong opinion on this, I leave it to the maintainers of the CI. I will add whatever you guys think is the best. Are you okay @jforissier with the suggestion?

File: core/drivers/crypto/caam/caam_jr.c [hunk 1]
```diff
@@ -5,6 +5,7 @@
  * Brief   CAAM Job Rings manager.
  *         Implementation of functions to enqueue/dequeue CAAM Job Descriptor
  */
+#include <caam_hal_clk.h>
```
[the remark this answers is not in the document]
Comment (clementfaure) [reply core/drivers/crypto/caam/caam_jr.c:8]: Ok

**Comment by jforissier** [comment]
> > `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with include ordering fixed in _caam_jr.c_.
> > For commit "ci: qemuv8: build with maximum log level", if willing to build with most debug features enabled, maybe consider (some of) the followings: `CFG_TEE_TA_MALLOC_DEBUG=y CFG_TEE_TA_MALLOC_DEBUG=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_CORE_STACK_PROTECTOR=y` `CFG_TA_STACK_PROTECTOR_ALL=y` `CFG_TA_STATS=y`.
> 
> Well, I don't have a strong opinion on this, I leave it to the maintainers of the CI. I will add whatever you guys think is the best. Are you okay @jforissier with the suggestion?

Let's leave it like this, it reproduces the same configuration as for QEMU (non-v8).

**Comment by clementfaure** [comment]
Force-push:
- Tags applied
- Rebase

Note: Checkpatch fail is expected for `587406a7d ta: pkcs11: fix trace compilation warning`

**Pull request merged**